cmd/zpool/zpool_main.c

```diff
@@ -2844,12 +3441,262 @@ get_timestamp_arg(char c)
 }
 
 /*
- * zpool iostat [-gLPv] [-T d|u] [pool] ... [interval [count]]
+ * Return stat flags that are supported by all pools.  "data" should be
+ * initialized to all 0xFFs before running.  It will get ANDed down until
+ * only the flags that are supported on all pools remain.
+ */
+static int
+get_stat_flags_cb(zpool_handle_t *zhp, void *data)
+{
+	uint64_t *mask = data;
+	nvlist_t *config, *nvroot;
+	uint64_t flags = 0;
+	int i, j;
+
+	/*
+	 * Lookup table for iostat flags to nvlist names.  Basically a list
+	 * of all the nvlists a flag requires.
+	 */
+	static const char *type_to_nvlist[IOS_COUNT][10] = {
+		[IOS_DEFAULT] = {ZPOOL_CONFIG_VDEV_STATS, NULL},
```

Comment:
I'd suggest putting each ZPOOL_CONFIG_* entry and the NULL on its own line to improve readability.  It helps eliminate any confusion when looking at this code.

```
                [IOS_DEFAULT] = {
                    ZPOOL_CONFIG_VDEV_STATS, 
                    NULL },
```